opentracing-proxy/opentracing-proxy/src/main/java/org/zalando/opentracing/proxy/plugin/LogCorrelation.java

```diff
@@ -75,9 +78,16 @@ public LogCorrelation withBaggage(final Map<String, String> mapping) {
 
     @Override
     public void onActivated(final Scope scope, final Span span) {
+        if (existingContextStack.get() == null)
```

Comment:
This conditional initialization can be removed in favor of:

```java
    private final ThreadLocal<Deque<Map<String, String>>> contexts =
            ThreadLocal.withInitial(LinkedList::new);
```

Please note, that this also changes the type from Stack to Deque.
Stack is synchronized which is overhead when there is exactly one per thread anyway.

Reply:
👍